controlplane/kubeadm/controllers/scale.go

```diff
@@ -121,7 +112,7 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: healthCheckFailedRequeueAfter}
 	}
 	// If etcd leadership is on machine that is about to be deleted, move it to the newest member available.
-	etcdLeaderCandidate := ownedMachines.Newest()
+	etcdLeaderCandidate := controlPlane.Machines.Newest()
```

Comment:
It'd have to be a weird scenario, like someone manually changing the KCP spec hash, to have the newest machine be the one being scaled down

Reply:
In case of manual machine deletion (user goes in an deletes a machine), or if in the future the last machine created is the one that we'll remediate because it's unhealthy